kernel/kernel-api/src/main/java/io/delta/kernel/PaginatedAddFilesIterator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.utils.CloseableIterator;
+
+/**
+ * An iterator over {@link FilteredColumnarBatch}, each representing a batch of Scan Files in a
+ * paginated scan. This iterator also exposes the page token that can be used to resume the scan
+ * from the exact position current page ends in a subsequent request.
+ *
+ * <p>This interface extends {@link CloseableIterator} and should be closed when the iteration is
+ * complete.
+ */
+public interface PaginatedAddFilesIterator extends CloseableIterator<FilteredColumnarBatch> {
```

Comment:
nit: we should not leak `AddFile`s. This is a PaginatedIterator over `ScanFile`s